x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -1557,14 +1580,21 @@ class AgentPolicyService {
     }
   }
 
-  private packagePoliciesWithoutMultiplePolicies(packagePolicies: PackagePolicy[]) {
+  private packagePoliciesWithSingleAndMultiplePolicies(packagePolicies: PackagePolicy[]): {
+    policiesWithSingleAP: PackagePolicy[];
+    policiesWithMultipleAP: PackagePolicy[];
+  } {
     // Find package policies that don't have multiple agent policies and mark them for deletion
     if (appContextService.getExperimentalFeatures().enableReusableIntegrationPolicies) {
```

Comment:
Hm you are right, I think we should add the check on backend too, and not allow sharing a package policy without enterprise licence in the API.

Reply:
I thought more about this, and I think it's not really good to check the feature flag/license when deleting an agent policy, because it will result in inconsistent state. E.g. if the license is degraded, and an agent policy is deleted, it will result in all shared integration policies to be deleted, which is a destructive action.

I think it's enough to prevent adding multiple agent policies with the license check.

Reply:
Added a follow up pr to add the license check to create/update package policy API: https://github.com/elastic/kibana/pull/187467